devlib/module/sched.py

```diff
@@ -469,12 +469,25 @@ def get_capacities(self, default=None):
         cpus = self.target.list_online_cpus()
 
         capacities = {}
-        sd_info = self.get_sd_info()
 
         for cpu in cpus:
             if self.has_dmips_capacity(cpu):
                 capacities[cpu] = self.get_dmips_capacity(cpu)
-            elif self.has_em(cpu, sd_info.cpus[cpu]):
+
+        missing_cpus = set(cpus).difference(capacities.keys())
+        if not missing_cpus:
+            return capacities
+
+        if not SchedProcFSData.available(self.target):
+            if default != None:
+                capacities.update({cpu : default for cpu in missing_cpus})
+                return capacities
+            else:
+                raise RuntimeError('No capacity data for cpus {}'.format(missing_cpus))
```

Comment:
It's a good idea to turn sets into sorted lists before leaking them outside, since the iteration order is not defined. This allows useful diff from outputs of one run to another.

Reply:
Right, let me fix and resend.